### ISSUE TITLE
Why not reference the local CSS variables in components for different states?

### ISSUE THREAD

**Issue by tahmid-stacker** [twbs/bootstrap#38360]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Let's use the primary button as an example. Right now, the code looks like this:

```css
.btn-primary {
	--bs-btn-color: #fff;
	--bs-btn-bg: #0d6efd;
	--bs-btn-border-color: #0d6efd;
	--bs-btn-hover-color: #fff;
	--bs-btn-hover-bg: #0b5ed7;
	--bs-btn-hover-border-color: #0a58ca;
	--bs-btn-focus-shadow-rgb: 49, 132, 253;
	--bs-btn-active-color: #fff;
	--bs-btn-active-bg: #0a58ca;
	--bs-btn-active-border-color: #0a53be;
	--bs-btn-active-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
	--bs-btn-disabled-color: #fff;
	--bs-btn-disabled-bg: #0d6efd;
	--bs-btn-disabled-border-color: #0d6efd;
}
```

What I'm proposing would refactor this into the following:

```css
.btn-primary {
	--bs-btn-color: #fff;
	--bs-btn-bg: #0d6efd;
	--bs-btn-border-color: #0d6efd;
	--bs-btn-hover-color: var(--bs-btn-color);
	--bs-btn-hover-bg: #0b5ed7;
	--bs-btn-hover-border-color: #0a58ca;
	--bs-btn-focus-shadow-rgb: 49, 132, 253;
	--bs-btn-active-color: var(--bs-btn-color);
	--bs-btn-active-bg: #0a58ca;
	--bs-btn-active-border-color: #0a53be;
	--bs-btn-active-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
	--bs-btn-disabled-color: var(--bs-btn-color);
	--bs-btn-disabled-bg: var(--bs-btn-bg);
	--bs-btn-disabled-border-color: var(--bs-btn-border-color);
}
```
In short, why not reference the existing local variables for the different states which share the same value? Are there any downsides to this approach that I'm not aware of?

### Motivation and context

The main motivation is pretty simple, when trying to customize a component, less variable values would need to be overridden, especially if the references make logical sense for the most part.

**Comment by mdo** [comment]
This is basically entirely because of Sass, but we want to change this in v6. Until then, we're limited by backward compatibility, which would be broken if we suddenly stopped applying the Sass value to certain properties. In some cases we're able to reassign things to their CSS variable equivalent, but it's not a given.
